bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/parser/CommonRpcParser.java

```diff
@@ -173,8 +173,14 @@ protected HmDatapoint assembleDatapoint(String name, String unit, String type, S
         }
 
         HmValueType valueType = HmValueType.parse(type);
-        if (valueType == null || valueType == HmValueType.UNKNOWN) {
+        if (valueType == null || valueType == HmValueType.UNKNOWN)
             throw new IOException("Unknown datapoint type: " + type);
+        else if (valueType == HmValueType.FLOAT && dp.getUnit() == null
+                && StringUtils.contains(dp.getName(), "TEMPERATURE")) {
```

Comment:
Can this be matched more exact than contains, what are possible values?

Reply:
There a names with TEMPERATURE at the end and also names with TEMPERATURE in the middle. All datapoints with possibly missing units have in common that they are defined as float and contain "TEMPERATURE". I could check the documentation whether it makes sense to limit it more.

Reply:
Would be appreciated :-)

Reply:
Replaced by regex that is more restrictive :-)